src/FsAutoComplete/CodeFixes/ConvertTripleSlashCommentToXmlTaggedDoc.fs

```diff
@@ -0,0 +1,217 @@
+module FsAutoComplete.CodeFix.ConvertTripleSlashCommentToXmlTaggedDoc
+
+open FsToolkit.ErrorHandling
+open FsAutoComplete.CodeFix.Types
+open Ionide.LanguageServerProtocol.Types
+open FsAutoComplete
+open FsAutoComplete.LspHelpers
+open FSharp.Compiler.Syntax
+open FSharp.Compiler.Text.Range
+open FSharp.Compiler.Xml
+open System
+
+let title = "Convert '///' comment to XML-tagged doc comment"
+
+let private containsPosAndNotEmptyAndNotElaborated (pos: FSharp.Compiler.Text.Position) (xmlDoc: PreXmlDoc) =
+  let d = xmlDoc.ToXmlDoc(false, None)
+
+  if rangeContainsPos d.Range pos then
+    let summaryPresent =
+      d.UnprocessedLines |> Array.exists (fun s -> s.Contains("<summary>"))
+
+    not xmlDoc.IsEmpty && not summaryPresent
```

Comment:
can you rearrange this to exit early when possible? so prefer not to call `.ToXmlDoc(false, None)` at all until the `not xmlDoc.IsEmpty` condition has been checked. Conceptually something like `not xmlDoc.IsEmpty && hasSummary xmlDoc`.

Reply:
Done